Added keyboard accessible shortcut menu for create modals

Blackshadow8910 opened the pull request:
<!--
  This template provides some ideas of things to include in your PR description.
  To start, try providing a short summary of your changes in the Title above.
  If a section of the PR template does not apply to this PR, then delete that section.
 -->

## What type of PR is this?

<!--
  Delete any of the following that do not apply:
 -->

- feature

## What this PR does / why we need it:

This PR adds a command palette like modal menu that can be brought up with Ctrl+`, and allows for faster access to the other create modals for keyboard only use. Other commands can be added later but I can't think of anything, so for now its just a keyboard shortcut to those menus. The shortcut should also probably be displayed somewhere.

<!--
  What goal is this change working towards?
  Provide a bullet pointed summary of how each file was changed.
  Briefly explain any decisions you made with respect to the changes.
  Include anything here that you didn't include in *Release Notes*
  above, such as changes to CI or changes to internal methods.
-->

I also changed `BaseModal.vue` to allow clicking off of the modal to close it and made the close button optional, since I think these additions benefit this feature.

## Which issue(s) this PR fixes:

#16 

<!--
If this PR fixes one of more issues, list them here.
One line each, like so:
Fixes #123
Fixes #39
-->

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Release Notes

- **New Features**
  - Introduced a configurable Modal component with an optional close button.
  - Added a new Quick Menu input interface for rapid action selection.
  - Implemented a Quick Menu modal with keyboard shortcut support (Ctrl + Backquote).

- **Improvements**
  - Enhanced modal component flexibility.
  - Improved user interaction with quick action selection.
  - Added keyboard navigation for modal and quick menu actions.

- **UI Changes**
  - Updated dropdown content class ordering.
  - Integrated Quick Menu modal into the default layout.
  - Added localization entries for quick menu guidance.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

tonyaellie commented on frontend/components/global/QuickMenu/Modal.vue:
```diff
@@ -0,0 +1,70 @@
+          </button>
+        </li>
+      </ul>
+      <span class="text-base-300">Use number keys to quick select.</span>
```
This text is currently kinda unreadable, I think it would be better if it was just the normal text colour?
![image](https://github.com/user-attachments/assets/41d7b4fd-e7a5-4a18-b3f9-63ee703e3720)
It also gets blocked off by the dropdown so moving it to be above would make it more readable.
![image](https://github.com/user-attachments/assets/cd3a67a2-b4c1-40dc-b72a-3ec86186815a)

tonyaellie commented on frontend/components/global/QuickMenu/Input.vue:
```diff
@@ -0,0 +1,108 @@
+      class="input input-bordered w-full"
+      @input="inputValue = $event.target.value"
+    ></ComboboxInput>
+    <ComboboxOptions class="card dropdown-content absolute w-full rounded-lg border border-base-300 bg-base-100">
```
Might be worth adding `mt-2` as currently it looks a bit off and that matches what we do in Autocomplete2.tsx

tonyaellie reviewed (changes requested):
Other things to consider:
Would be nice to have a <kbd>Ctrl + `</kbd> by the create button to help new users spot the shortcut, maybe hide it on mobile but that isnt very necesarry.
![image](https://github.com/user-attachments/assets/90f03fbe-7a1c-45b0-873d-24b4eefef47d)
Having hotkeys to directly open the create items, create labels and create create locations could be nice though not a must have.

Translation keys need to added.

Blackshadow8910 commented:
I've added translation keys, a tooltip indicating the shortcut, and fixed the text in the modal. I do think the text looks pretty ugly and it would be better if it weren't there, but I don't know if the hotkeys are obvious enough to not include it.

Direct hotkeys for the create modals would probably be good but I just don't know what shortcuts would make sense. The browser takes up most of the sensible key combinations. I wanted to add customizable hotkeys but 1. for the same reason there aren't that many options and 2. I don't know how.

tonyaellie reviewed (changes requested):
Looks good, just need to add any new translation keys to the https://github.com/sysadminsmedia/homebox/blob/main/frontend/locales/en.json

Blackshadow8910 commented:
Ah, sorry, I forgot to commit the file

katosdev commented:
LGTM! Thanks for the PR.
Just because I've broken enough things of late ( 👀 ) I'll just ask a quick sanity check of @tankerkiller125 please 🙂

mcarbonne commented:
This PR introduced a bug: it is not possible anymore to pick a date when editing a maintenance entry.
Can anyone confirm that I'm not the only one concerned ?
(I'm using firefox and after reverting this PR in my fork, it's working again).

Blackshadow8910 commented:
Ok, yeah, sorry, that's because of the change to modals allowing them to close when clicks are made outside of them. I reverted this change for all modals other than the new one in #467 so that should fix it.
